runtime/compiler/env/DependencyTable.cpp

```diff
@@ -469,9 +470,32 @@ TR_AOTDependencyTable::resolvePendingLoads()
    _pendingLoads.clear();
    }
 
-TR_OpaqueClassBlock *
-TR_AOTDependencyTable::findClassCandidate(uintptr_t romClassOffset)
+J9Class *
+TR_AOTDependencyTable::findCandidateWithChainAndLoader(TR::Compilation *comp, uintptr_t classChainOffset, void *classLoaderChain)
+   {
+   if (comp->isDeserializedAOTMethod() || comp->ignoringLocalSCC())
+      return NULL;
+
+   auto classChain = (uintptr_t *)_sharedCache->pointerFromOffsetInSharedCache(classChainOffset);
+   return findChainLoaderCandidate(comp, classChain, classLoaderChain);
+   }
+
+J9Class *
+TR_AOTDependencyTable::findCandidateWithChainAndLoader(TR::Compilation *comp, uintptr_t *classChain, void *classLoaderChain)
+   {
+   if (comp->isDeserializedAOTMethod() || comp->ignoringLocalSCC())
+      return NULL;
+
+   return findChainLoaderCandidate(comp, classChain, classLoaderChain);
+   }
+
+J9Class *
+TR_AOTDependencyTable::findChainLoaderCandidate(TR::Compilation *comp, uintptr_t *classChain, void *classLoaderChain)
    {
+   TR_ASSERT(classLoaderChain, "Must be given a loader chain");
```

Comment:
Best to just make this a fatal assert; there's not really a good reason for adding new asserts as `TR_ASSERT`.